src/test/components/ButtonWithPanel.test.js

```diff
@@ -52,6 +52,49 @@ describe('shared/ButtonWithPanel', () => {
     expect(container.firstChild).toMatchSnapshot();
   });
 
+  describe('various panel contents', () => {
+    it('renders panels with default buttons and titles', () => {
+      const { container } = render(
+        <ButtonWithPanel
+          className="button"
+          label="My Button"
+          open={true}
+          panel={
+            <ArrowPanel
+              className="panel"
+              title="Wonderful content"
+              okButtonText="Confirm"
+              cancelButtonText="Cancel"
+            >
+              <div>Panel content</div>
+            </ArrowPanel>
+          }
+        />
+      );
+      expect(container.firstChild).toMatchSnapshot();
+    });
+
+    it('renders panels with specified buttons', () => {
+      const { container } = render(
+        <ButtonWithPanel
+          className="button"
+          label="My Button"
+          open={true}
+          panel={
+            <ArrowPanel
+              className="panel"
+              okButtonText="Confirm"
```

Comment:
(observation) I'm somewhat surprised you are using these built-in ones and didn't just strip them out. They seem pretty specific to the original use-case of the component, back in the original Markus days. It seems like having some individual components for them would be better that are separate from the arrow panel.

Reply:
Yeah, I thought of that indeed. The rationale I decided is that with more opinionated and stricter components we could have more standardized styling throughout the application with less effort. I'm not completely convinced so we can still possibly revisit in the future if needed.